ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoSignedTransaction.kt

```diff
@@ -0,0 +1,54 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata
+import net.corda.v5.application.serialization.SerializationService
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+import java.security.PublicKey
+
+/**
+ * Defines a signed UTXO transaction.
+ *
+ * @property id The ID of the transaction.
+ * @property signatures The signatures that have been applied to the transaction.
+ */
+@DoNotImplement
+@CordaSerializable
+interface UtxoSignedTransaction {
+
+    val id: SecureHash
+    val signatures: Set<DigitalSignatureAndMetadata>
```

Comment:
Consensual friend of this class has some differences:
* signatures is a List<>
* toLedgerTransaction does not have the serializer as an argument. (It will get injected as a constructor parameter) (The same for other methods where serializer will be needed.)

Reply:
It sounds convincing.
I think the main reason for `List<>` in consensual was only to follow the C4 pattern.
If @mnesbit or @blsemo are happy with `Set<>`, I will change the consensual signed Tx as well.

Reply:
Remember that for RSA and ECDSA signatures repeat signing will produce different values as the algorithm includes entropy. (This isn't true for EdDSA that uses a cryptographic trick to remove need for the entropy and be deterministic, which is lovely but we can't generalise)